packages/SwingSet/test/bootstrap-relay.js

```diff
@@ -55,10 +63,49 @@ export const buildRootObject = () => {
       // braces to unharden so it can be hardened
       encodePassable(
         Far(label, {
-          ...objectMap(methodReturnValues, v => () => decodePassable(v)),
+          ...objectMap(methodReturnValues, v => {
+            const ret = decodePassable(v);
+            return () => ret;
+          }),
         }),
       ),
 
+    /**
+     * Does the same as makeSimpleRemotable, but also logs the method and
+     * arguments, accessible by calling `getLogForRemotable(remotable)`.
+     *
+     * @param {string} label
+     * @param {Record<string, any>} methodReturnValues
+     */
+    makeLoggingRemotable: (label, methodReturnValues) => {
+      const logs = [];
+      const remotable = Far(label, {
+        ...objectMap(methodReturnValues, (v, k) => {
+          const ret = decodePassable(v);
+          return (...args) => {
+            logs.push([k, ...args]);
```

Comment:
I clarified this class of problems.  Basically, the encoding/decoding was messed up if you called bootstrap methods directly instead of via the `messageVatObject` intermediary.  I changed the `makeRemotable` path not to do its own encode/decode and declare the invariant that `messageVatObject` should be used (which does it once upon entry and exit outside the kernel).

Reply:
I don't think that solution makes sense; **all** methods of the object exported by bootstrap-relay `buildRootObject` are intended to be called directly by testing code with control over the vat environment and all arguments and return values should therefore be subject to translation. That translation _should_ be less kludgy and more in line with analogous vat-to-kernel-to-vat ref management, but haven't yet had time for effecting such a change.

Reply:
Okay, I can agree with this argument for making `makeRemotable` and `getLogForRemotable` do the appropriate decode/encode mapping.

However, I disagree with your suggested change.  I send the constructed Remotable into other SwingSet vats.  Because of this, the `args` in question (i.e. the arguments of a method on the Remotable) come from within SwingSet and thus should *not* be `decodePassable`d.  Only when they are returned by `getLogForRemotable`, should they be `encodePassable`d since they are leaving SwingSet.

Reply:
Got it. I accept the premise, and will add a ````` ```suggestion ````` to that effect. And hopefully the `encodePassable`/`decodePassable` infrastructure can be removed completely in the relatively near future, because I think `kunser`/`kser`/etc. can actually handle the job.